### PR TITLE
WT-7319 Implement a checkpoint_manager class that creates checkpoints at configurable intervals

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
This PR adds a new component to call a checkpoint repeatedly. We can use the existing throttle capabilities to configure the interval.

### CODE REVIEW

**Comment by luke-pearson** [comment]
This is essentially what I expected, the op_count and interval does look a bit weird to me now so I think we'll create a ticket to make it a single string called rate or similar with "5s".

**Review by luke-pearson — APPROVED** [review]
LGTM.

File: test/cppsuite/test_harness/checkpoint_manager.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    {
+    }
+    virtual ~checkpoint_manager() = default;
+
+    /* Delete the copy constructor and the assignment operator. */
+    checkpoint_manager(const checkpoint_manager &) = delete;
+    checkpoint_manager &operator=(const checkpoint_manager &) = delete;
+
+    void
+    load() override
```
Comment (etienneptl) [inline test/cppsuite/test_harness/checkpoint_manager.h:50]: Would be great to use the override keyword in the other derived classes too. We could probably address it in a dedicated ticket or when we get the chance to work on those files in other tickets.

**Review by etienneptl — APPROVED** [review]
LGTM !

File: test/cppsuite/test_harness/checkpoint_manager.h (hunk 1 above)
Comment (tetsuo-cpp) [reply test/cppsuite/test_harness/checkpoint_manager.h:50]: Definitely. It gives us a bit of extra safety from the compiler. 👍

**Pull request merged**